core/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -201,6 +201,15 @@ public void writeOptionalVInt(@Nullable Integer integer) throws IOException {
         }
     }
 
+    public void writeOptionalFloat(@Nullable Float aFloat) throws IOException {
+        if (aFloat == null) {
+            writeBoolean(false);
+        } else {
+            writeBoolean(true);
+            writeFloat(aFloat);
+        }
+    }
+
```

Comment:
`writeGenericValue` also works on Strings etc .. and also handles optionals. This is more for readable. Question: should we use the generic method all the time, or its writeX counter part?